sunbeam/parser.cpp

```diff
@@ -12,14 +12,21 @@ namespace {
     Deck create_deck( const std::string& deckStr,
                       const ParseContext& pc,
                       const Parser& parser) {
-        return parser.parseFile( deckStr, pc);
+
+        Opm::ErrorGuard guard;
+        auto p = parser.parseFile( deckStr, pc, guard );
+        guard.clear();
```

Comment:
Observe that the `ErrorGuard` only plays in the case there are errors which have been configured to be handled with the `DELAYED_EXIT1` behavior - that is by default none, and I do not see that as changing. In the `flow` application there is a commandline switch `--strict-parse=true` which will set all error handling behavior to `DELAYED_EXIT1` - and then the `ErrorGuard`will come into play.

*iff* there are errors you should not just return the deck; the intended usage is either to let it just dwindle down through the destructor (I tried to throw from the destructor - but that became `std::terminate` - whatever I did), or alternatively something like this:
```C++
if (guard) {
    // Dump errors and warnings on stderr
   guard.dump();
   guard.clear();
   throw std::runtime_error("Fatal errors were encountered while parsing");
}
```